src/coreclr/jit/morphblock.cpp

```diff
@@ -1259,9 +1259,9 @@ GenTree* MorphCopyBlockHelper::CopyFieldByField()
                     }
                     else
                     {
-                        if (i == 0)
+                        if (i == (fieldCnt - 1))
                         {
-                            // Use the orginal m_dstAddr tree when i == 0
+                            // Reuse the orginal m_dstAddr tree for the last field.
```

Comment:
Should we assert that we're not reusing an address with a zero offset field seq?

Reply:
Hmm, it is legal for the address to contain a zero-offset sequence before the morphing, so I suppose asserting here would require:

1) Getting the "base" sequence (essentially perform the reverse of what `fgAddFieldSeqForZeroOffset` does).
2) After morphing, loop over all the created address trees and assert that we only added one field to the "base", and all of them are distinct.

I am not sure it is worth the amount of code required. Once #58312 is finished, we will have asserts in VN that would have caught this case.

Reply:
I'm not a fan of how we handle the zero offset case today. We've had a number of bugs in this area.

Ok by me if it is too hard to check -- just wanted to see if you had any ideas.